### ISSUE TITLE
into datetime doesnt't work with --format and ignores user's locale

### ISSUE THREAD

**Issue by orr721** [nushell/nushell#11015]
### Describe the bug

The format parsing rules in **into datetime --format** do not work and result in an error.

The flexible parsing rules (without --format) ignore user's locale and only use US format.

Also the text of the error description is wonky.

Thanks!

### How to reproduce

Case 1. Parsing fails with -f
```
~> "06.03.2023" | into datetime -f "%d.%m.%Y"
Error: nu::shell::cant_convert

  × Can't convert to could not parse as datetime using format '%d.%m.%Y'.
   ╭─[entry #28:1:1]
 1 │ "06.03.2023" | into datetime -f "%d.%m.%Y"
   ·                ──────┬──────
   ·                      ╰── can't convert input is not enough for unique date and time to could not parse as datetime using format '%d.%m.%Y'
   ╰────
  help: you can use `into datetime` without a format string to enable flexible parsing
```

Case 2. Flexible parsing uses US locale only
```
~> date now
Thu, 9 Nov 2023 15:25:35 +0100 (now)
~> date now | format date "%d.%m.%Y" | into string | into datetime
Mon, 11 Sep 2023 00:00:00 +0100 (2 months ago)
```

### Expected behavior

Case 1.
`"06.03.2023" | into datetime -f "%d.%m.%Y"`
should parse correctly into 6 March 2023 without an error

Case 2.
`date now | format date "%d.%m.%Y" | into string | into datetime`
should parse to currend date respecting user's locale

### Configuration

| key                | value                                                                                       |
| ------------------ | ------------------------------------------------------------------------------------------- |
| version            | 0.86.0                                                                                      |
| branch             |                                                                                             |
| commit_hash        |                                                                                             |
| build_os           | linux-x86_64                                                                                |
| build_target       | x86_64-unknown-linux-gnu                                                                    |
| rust_version       | rustc 1.73.0 (cc66ad468 2023-10-03)                                                         |
| rust_channel       | stable-x86_64-unknown-linux-gnu                                                             |
| cargo_version      | cargo 1.73.0 (9c4383fb5 2023-08-26)                                                         |
| build_time         | 2023-10-19 14:48:50 +02:00                                                                  |
| build_rust_channel | release                                                                                     |
| allocator          | mimalloc                                                                                    |
| features           | default, sqlite, trash, which, zip                                                          |
| installed_plugins  | from eml, from ics, from ini, from vcf, gstat, inc, query, query json, query web, query xml |

### Additional context

```
~> locale
LANG=sk_SK.UTF-8
LANGUAGE=
LC_CTYPE="sk_SK.UTF-8"
LC_NUMERIC="sk_SK.UTF-8"
LC_TIME="sk_SK.UTF-8"
LC_COLLATE="sk_SK.UTF-8"
LC_MONETARY="sk_SK.UTF-8"
LC_MESSAGES="sk_SK.UTF-8"
LC_PAPER="sk_SK.UTF-8"
LC_NAME="sk_SK.UTF-8"
LC_ADDRESS="sk_SK.UTF-8"
LC_TELEPHONE="sk_SK.UTF-8"
LC_MEASUREMENT="sk_SK.UTF-8"
LC_IDENTIFICATION="sk_SK.UTF-8"
LC_ALL=

```

**Comment by fdncred** [comment]
I could be making this up, but from what I remember, using `into datetime` this way requires a time and timezone like:
```
"06.03.2023T00:00:00+0000" | into datetime --format "%d.%m.%YT%H:%M:%S%z"
```

**Comment by orr721** [comment]
Yes, I can understand the Case 2 is not working, most software is US centric anyway, we Europeans got used it..

But why is the parsing with the format string not working?

**Comment by fdncred** [comment]
> But why is the parsing with the format string not working?

Because it works but it requires time and timezone.

**Comment by orr721** [comment]
Isn't it unnecessarily strict then?  And probably it would be helpful to mention this in the help text as well.

Oh well, so I will need to parse the string manually all the time to the %F format and then use the "flexible" `into datetime`  to convert it to the datetime type.

**Comment by orr721** [comment]
I mean it just seems really convoluted to do it manually if the logic is already there..
```
~> "03.10.2023" | parse "{d}.{m}.{y}" | get 0 | $"($in.y)-($in.m)-($in.d)" | into datetime
Tue, 3 Oct 2023 00:00:00 +0100 (a month ago)

```

**Comment by orr721** [comment]
Uhh, ok, or this works too..
```
~> "03.10.2023" | $in + "T00:00:00Z+0000" | into datetime -f "%d.%m.%YT%TZ%z"
Tue, 3 Oct 2023 00:00:00 +0000 (a month ago)

```

**Comment by fdncred** [comment]
right, that's kind of how i did it
```
"06.03.2023" | $"($in)T00:00:00+0000" | into datetime --format "%d.%m.%YT%H:%M:%S%z"
```
and you're right, it is a little convoluted but it's what we have right now.

**Comment by agirardeau** [comment]
Isn't the root problem here that nushell doesn't have a separate `date` type, only `datetime`? You should be able to create a date value without an attached time, like in all other languages. This seems to be a common gotcha, see https://github.com/nushell/nushell/issues/3205, https://github.com/nushell/nushell/issues/9916, and https://github.com/nushell/nushell/pull/3207.

**Comment by fdncred** [comment]
closed by https://github.com/nushell/nushell/pull/15544
